documentation/ArrayComposition.md

```diff
@@ -0,0 +1,130 @@
+<!--
+Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+-->
+
+This note attempts to describe the motivation for and design of an
+implementation of Fortran 90 (and later) array expression evaluation that
+minimizes the use of dynamically allocated temporary storage for
+the results of calls to transformational intrinsic functions.
+
+The transformational intrinsic functions of Fortran of interest to
+us here include:
+
+* Reductions to scalars (`SUM(X)`, also `ALL`, `ANY`, `COUNT`, `IALL`,
```

Comment:
If we want to include F2008 features then there's `NORM2`

Reply:
Do you think that a `NORM2` can/should be computed inline in a compiler-generated loop instead of in a runtime library function?

Reply:
Performance wise it could better inline. However, actually implementing it inline might be convoluted and may not necessarily be the best solution. 
Is this document only talking about inline-able intrinsics?

Reply:
Yes, the topic is how to implement some transformational intrinsic functions in generated code rather than in a runtime library.  I can see how a naive `NORM2` as `SQRT(SUM(X**2))` would work inline, but not an overflow-avoiding algorithm as recommended by the standard.

Reply:
Agreed.